configure/CMakeLists.txt

```diff
@@ -31,30 +45,102 @@ else(OMNI_BASE)
-    set(ZMQ_PKG_LIBRARIES "-lzmq")
-    include_directories(${ZMQ_BASE}/include)
-    link_directories(${ZMQ_BASE}/lib)
+    message("Using ZMQ_BASE=$ENV{ZMQ_BASE}")
+    set(ZMQ_PKG_INCLUDE_DIRS $ENV{ZMQ_BASE}/include)
+    if (WIN32)
+        link_directories($ENV{ZMQ_BASE}/lib/Debug)
+        link_directories($ENV{ZMQ_BASE}/lib/Release)
+        link_directories($ENV{ZMQ_BASE}/bin/Debug)
+        link_directories($ENV{ZMQ_BASE}/bin/Release)
+        if (MSVC90)
+            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v90-mt-sgd-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v90-mt-gd-4_0_5.lib")
+            else()
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v90-mt-s-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v90-mt-4_0_5.lib")
+            endif()
+        endif()
+        if (MSVC10)
+            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v100-mt-sgd-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v100-mt-gd-4_0_5.lib")
+            else()
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v100-mt-s-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v100-mt-4_0_5.lib")
+            endif()
+        endif()
+        if (MSVC12)
+            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v120-mt-sgd-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v120-mt-gd-4_0_5.lib")
+            else()
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v120-mt-s-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v120-mt-4_0_5.lib")
+            endif()
+        endif()
+        if (MSVC14)
+            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v140-mt-sgd-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v140-mt-gd-4_0_5.lib")
+            else()
+                set(ZMQ_PKG_LIBRARIES_STA "libzmq-v140-mt-s-4_0_5.lib")
+                set(ZMQ_PKG_LIBRARIES_DYN "libzmq-v140-mt-4_0_5.lib")
+            endif()
+        endif()
+    else()
+        set(ZMQ_PKG_LIBRARIES "-lzmq")
+        link_directories(${ZMQ_BASE}/lib)
+    endif()
+    include_directories($ENV{ZMQ_BASE}/include)
 else(ZMQ_BASE)
     pkg_search_module(ZMQ_PKG REQUIRED libzmq)
     if(NOT ZMQ_PKG_FOUND)
         message(FATAL "ZMQ library is required for Tango")
     endif()
 endif(ZMQ_BASE)
 
+if(PTHREAD_WIN)
+    message("Using PTHREAD_WIN=$ENV{PTHREAD_WIN}")
+    add_definitions(-DHAVE_PTHREAD_NP_H)
+    set(PTHREAD_WIN_PKG_INCLUDE_DIRS $ENV{PTHREAD_WIN}/include)
+    
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(PTHREAD_WIN_PKG_LIBRARIES_DYN "pthreadVC2d.lib")
+        set(PTHREAD_WIN_PKG_LIBRARIES_STA "pthreadVC2-sd.lib")
+    else()
+        set(PTHREAD_WIN_PKG_LIBRARIES_DYN "pthreadVC2.lib")
+        set(PTHREAD_WIN_PKG_LIBRARIES_STA "pthreadVC2-s.lib")
+    endif()
+else(IDL_BASE)
```

Comment:
This else looks like a copy/paste error and actually breaks something on Linux in some cases

Reply:
clearly a bad copy